fsspec/dircache.py

```diff
@@ -0,0 +1,50 @@
+from functools import lru_cache
+import time
+
+
+class DirCache(dict):
```

Comment:
Since you're changing the `__init__` signature, I think it'd be better to compose this class with a `._cache` method rather than inheriting from dict. You might inherit from MutableMapping instead.

Reply:
I'll probably inherit from nothing in the end, since I don't expect calls to `keys()`, etc.